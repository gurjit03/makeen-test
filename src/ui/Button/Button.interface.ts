type Variant = 'contained'| 'outlined'
type Color = 'primary' | 'secondary'

export interface Button {
    variant: Variant;
    color: Color;
    onClick?: () => {};
    marginBottom?: number;
    children: React.ReactChild
}
