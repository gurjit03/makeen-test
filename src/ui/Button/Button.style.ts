import styled, { css } from 'styled-components'
import { colors, verticalSpacing } from '../../config/jss'
import { Button as ButtonProps } from './Button.interface'

const variantOptions: any = {
	contained: {
		primary: {
			backgroundColor: colors.primary,
			color: colors.white,
			borderColor: 'transparent',
		},
		secondary: {
			backgroundColor: colors.secondary,
			color: colors.white,
			borderColor: 'transparent',
		},
	},
	outlined: {
		primary: {
			backgroundColor: colors.white,
			borderColor: colors.primary,
			color: colors.primary,
		},
		secondary: {
			backgroundColor: colors.white,
			borderColor: colors.secondary,
			color: colors.secondary,
		},
	},
}

export const Wrapper = styled.button<ButtonProps>`
  padding: ${verticalSpacing}px;
  margin: 0;
  border: none;
  outline: none;
  border: 2px solid;
  border-radius: ${verticalSpacing}px;
  display: inline-block;
  margin-bottom: ${props => props.marginBottom || 0}px
  &:hover {
    cursor: pointer;
  }
   ${({ variant, color: buttonColor }) =>
				variant &&
				variantOptions[variant] &&
				css`
					background-color: ${variantOptions[variant][buttonColor].backgroundColor};
					color: ${variantOptions[variant][buttonColor].color};
					border-color: ${variantOptions[variant][buttonColor].borderColor};
				`}
`
