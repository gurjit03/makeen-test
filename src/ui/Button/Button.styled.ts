import styled, {css} from 'styled-components';
import { colors, verticalSpacing } from '../../config/jss';
import { Button as ButtonProps } from './Button.interface';


const variantOptions = {
    contained: {
        primary: {
            backgroundColor: colors.primary,
            color: colors.white
        },
        secondary: {
            backgroundColor: colors.secondary,
            color: colors.white
        }
    },
    outlined: {
        primary: {
            backgroundColor: colors.white,
            borderColor: colors.primary,
            color: colors.primary
        },
        secondary: {
            backgroundColor: colors.white,
            borderColor: colors.secondary,
            color: colors.secondary
        }
    }
}

export const Wrapper = styled.button<ButtonProps>`
  padding: ${verticalSpacing}px;
  margin: 0;
  border: none;
  outline: none;
  border: 2px solid;
  border-radius: ${verticalSpacing}px;
  display: inline-block;
  margin-bottom: ${props => props.marginBottom || 09}px
  &:hover {
    cursor: pointer;
  }
   ${({ variant }) =>
    variant &&
    variantOptions[variant] &&
    css`
       background-color: ${variantOptions[variant].backgroundColor};
       color: ${variantOptions[variant].color};
       border-color: ${variantOptions[variant].borderColor}
   `}
`;
