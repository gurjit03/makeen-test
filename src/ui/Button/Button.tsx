import React from 'react'
import { Wrapper } from './Button.style'
import { Button as ButtonProps } from './Button.interface'

const Button: React.FC<ButtonProps> = props => {
	const { children, ...rest } = props
	return <Wrapper {...rest}>{children}</Wrapper>
}

export default Button
