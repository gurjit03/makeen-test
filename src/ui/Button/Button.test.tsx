import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from './Button'


describe('<Button />', () => {
    it('should render properly', () => {
        const { asFragment } = render(<Button variant="contained" color="primary" >button</Button>)
        expect(asFragment()).toMatchSnapshot();
    });

    it('should display the content properly', () => {
        render(<Button variant="contained" color="primary" >button</Button>)
        expect(screen.getByText('button')).toBeInTheDocument()
    })
})

