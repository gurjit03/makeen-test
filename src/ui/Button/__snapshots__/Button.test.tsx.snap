// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBQB irKrPi"
    color="primary"
  >
    button
  </button>
</DocumentFragment>
`;
